### DISCUSSION TITLE
Cannot seem to mock a class that has the @InjectRepository by mikro-orm or because of its generic typing

### DISCUSSION THREAD

**Discussion by rienheuver** [automock/automock#268] [Q&A] [answered]
I have the following code in my test:
```ts
const { unit, unitRef } = TestBed.create(MyService)
  .compile();

myService = unit;
myRepository = unitRef.get(EntityRepository<MyEntity>);
```
and the following code shows the constructor of `MyService`:
```ts
constructor(
    @InjectRepository(MyEntity)
    private myRepository: EntityRepository<MyEntity>,
  ) {}
  ```

I currently get an `IdentifierNotFoundError` on the last line of the above code. I'm not sure if it's not working because of the `@InjectRepository` or because of the generic typing `<MyEntity>`. Is there a different way of going about this to make it work? I already tried `.mock().using()` but it will simply ignore the mock entirely (no error or warning given)

**Comment by omermorad** [comment, marked as answer]
Are you using NestJS as your framework?
Try use this:

```typescript
const { unit, unitRef } = TestBed.create(MyService).compile();
myService = unit;
myRepository = unitRef.get(getRepositoryToken(EntityRepository) as string);
```

Let's me know if it worked.

**Reply by rienheuver** [reply]
It did! Thanks :smile:
